cid.go

```diff
@@ -589,6 +595,13 @@ func (p Prefix) Sum(data []byte) (Cid, error) {
 	}
 }
 
+// BuildLink returns a new Link value based on the given hashsum,
+// and allows Prefix to satisfy the LinkPrototype interface.
+func (p Prefix) BuildLink(hashsum []byte) Cid {
+	c, _ := p.Sum(hashsum)
```

Comment:
panic if there's an error, instead? otherwise latent bugs could go unnoticed

Reply:
I think this does what you want, in that if an invalid - (i guess i'm thinking like `[]byte{}` but even that is valid?) sum is given the response is Cid.Undef which can be matched as an error case still

Reply:
Ah, I see. I think that warrants a mention in the godoc, or at least a comment above the line that ignores the error. If I was reading this function declaration alone, I'd have guessed there was a bug :)

Reply:
extended comment